### PR TITLE
fix(message): disable save on empty message

### PR DESCRIPTION
**Pull request by jasonwoodland**
**What this PR does** 📖
- Prevents saving an edited message as an empty message
- Fixes styling and locale strings in the edit message UI

**Which issue(s) this PR fixes** 🔨
AP-1631

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

**Review by josephmcg — CHANGES_REQUESTED** [review]
I like the solution. Could you remove `$data` and `$props` mentions and fix intellisense in the file? The computed need return types. Then you can remove the declaration stuff (screenshot below).

Please make a habit of fixing this whenever you see it

image

File: components/views/chat/message/edit/Edit.vue [hunk 1]
```diff
@@ -40,19 +31,22 @@ export default Vue.extend({
...
-    this.$data.content = this.$props.message
+    this.content = this.message
   },
   methods: {
     saveMessage() {
-      this.$emit('commitMessage', this.$data.content.slice(0, this.maxChars))
+      if (this.content.length === 0) {
+        return
+      }
+      this.$emit('commitMessage', this.content.slice(0, this.maxChars))
```
Comment (WanderingHogan) [inline components/views/chat/message/edit/Edit.vue:49]: why not just `if (this.content.length > 0) this.$emit('commitMessage', this.content.slice(0, this.maxChars))`

**Review by WanderingHogan — APPROVED** [review]

File: components/views/chat/message/edit/Edit.vue (hunk 1 above)
Comment (josephmcg) [reply components/views/chat/message/edit/Edit.vue:49]: I believe the current way falls more in line with our 'guard clause' philosophy. although I'm lazy, I would write `!this.content.length`

**Review by josephmcg — APPROVED** [review]

**Review by stavares843 — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/29093946/171073731-b87a9f08-67de-42c3-828e-f1487283bf13.mov

if you try to save a empty message doesn't return any feedback to the user when clicking on save

**Comment by josephmcg** [comment]
true. error toast would be enough.
a nice bonus feature would be making the blue outline red if error, that would be fairly involved though

**Review by stavares843 — APPROVED** [review]

**Pull request merged**